runatlantis.io/docs/terraform-versions.md

```diff
@@ -24,6 +24,6 @@ terraform {
 See [Terraform `required_version`](https://www.terraform.io/docs/configuration/terraform.html#specifying-a-required-terraform-version) for reference.
 
 ::: tip NOTE
-Atlantis will automatically download the version specified.
+Atlantis will automatically download the version specified into `/usr/local/bin/terraform${ATLANTIS_TERRAFORM_VERSION}`. eg. `/usr/local/bin/terraform0.11.14`
```

Comment:
It only downloads the `--default-tf-version` binary there. The binaries set via terraform config or atlantis.yaml get downloaded to `$HOME/.atlantis/bin/terraform${VERSION}`

Reply:
@lkysow hm, we are running atlantis in fargate, so it makes it a bit tricky to get in and poke around, but after upgrading to atlantis 0.12, we had to make the following change to our custom workflows to make it work for both TF 0.11 and 0.12 binaries:

![](https://s3.amazonaws.com/mwarkentin-snaps/make_atlantis__waveaccountingkellerde21747_2020-05-01_08-38-09.png)

Do you have a pointer to where the code is for the binary downloads - just trying to square what you're saying with what we were seeing.

Reply:
@lkysow I think it's because of this: https://github.com/runatlantis/atlantis/blob/master/Dockerfile#L9-L16

We're using 0.11.14 and 0.12.24 so both of those are picked up by default in `/usr/local/bin` it looks like..

Reply:
Also, I just noticed this in the changelog for 0.9:

* The directory where Atlantis downloads Terraform binaries is now in the PATH of custom workflows (#678)

So maybe our problem goes away if we just drop the absolute path from the workflow

Reply:
> So maybe our problem goes away if we just drop the absolute path from the workflow

yes that should work now since we made that change.

There's 3 sources for tf downloads:
1. Docker image puts them in /usr/local/bin
1. --default-tf-version puts it in ~/.atlantis/bin
1. atlantis.yaml or terraform_version config on individual projects puts in in ~/.atlantis/bin